zip-0032.rst

```diff
@@ -361,6 +425,51 @@ Orchard child key derivation
 - Use :math:`I_L` as the child spending key :math:`\mathsf{sk}_{i}`.
 - Use :math:`I_R` as the child chain code :math:`\mathsf{c}_i`.
 
+Orchard internal key derivation
+-------------------------------
+
+As in the case of Sapling, for each external address, we want to produce another address
+for use by wallets for internal operations such as change and auto-shielding. That is, for
+each external full viewing key we need to be able to derive a single internal full viewing
+key that has viewing authority for just internal transfers. We also need to be able to
+derive the corresponding internal spending key if we have the external spending key.
+
+Let :math:`\mathsf{ask}` be the spend authorizing key if available, and
+let :math:`(\mathsf{ak}, \mathsf{nk}, \mathsf{rivk})` be the corresponding external full
+viewing key, obtained as specified in [#protocol-orchardkeycomponents]_.
+
+- Let :math:`K = \mathsf{I2LEBSP}_{256}(\mathsf{rivk})`.
+- Let :math:`\mathsf{rivk_{internal}} = \mathsf{ToScalar^{Orchard}}(\mathsf{PRF^{expand}}(K, [\mathtt{0x83}] \,||\, \mathsf{I2LEOSP_{256}}(\mathsf{ak}) \,||\, \mathsf{I2LEOSP_{256}}(\mathsf{nk}))`.
+
+Then the expanded internal spending key is :math:`(\mathsf{ask}, \mathsf{nk}, \mathsf{rivk_{internal}})`,
+and the internal full viewing key is :math:`(\mathsf{ak}, \mathsf{nk}, \mathsf{rivk_{internal}})`.
+
+Unlike `Sapling internal key derivation`_, we do not base this internal key derivation
+procedure on non-hardened derivation, which is not defined for Orchard. We can obtain the
+desired separation of viewing authority by modifying only the :math:`\mathsf{rivk_{internal}}`
+field relative to the external full viewing key, which results in different
+:math:`\mathsf{dk_{internal}}`, :math:`\mathsf{ivk_{internal}}` and :math:`\mathsf{ovk_{internal}}`
+fields being derived, as specified in [#protocol-orchardkeycomponents]_.
+
+The values of :math:`K` and :math:`\mathsf{rivk_{internal}}` are the same between deriving
+a full viewing key, and deriving the corresponding spending key. Both of these derivations
+are shown in the following diagram:
+
+.. figure:: zip-0032-orchard-internal-key-derivation.png
+    :width: 720px
+    :align: center
+    :figclass: align-center
+
+    Diagram of Orchard internal key derivation, also showing derivation from the parent extended spending key
+
+Note that there does not exist an :math:`\mathsf{sk}` that directly generates the internal key
+components by the procedure described in [#protocol-orchardkeycomponents]_. A similar effect
+could be obtained by storing a flag that indicates external or internal usage along with the
+original :math:`\mathsf{sk}`, but we do not define a specific format for this.
+
```

Comment:
@str4d says that this note isn't necessary if we deemphasize (ask, nk, rivk) and merge the two sides of the diagram at the spending key level.